src/compositemark/boxplot.ts

```diff
@@ -165,86 +160,193 @@ export function normalizeBoxPlot(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT
   };
 }
 
-export function box2DOrient(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT | BoxPlotDef>): Orient {
+export function boxOrient(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT | BoxPlotDef>): Orient {
   const {mark: mark, encoding: encoding, ...outerSpec} = spec;
 
-  // FIXME: refactor code such that we don't have to do this casting
-  // We can cast here as we already check from outside that both x and y are FieldDef
-  const xDef = encoding.x as FieldDef<Field>;
-  const yDef = encoding.y as FieldDef<Field>;
-  let resultOrient: Orient;
+  const is2D = encoding.x && encoding.y;
```

Comment:
Instead of branching all sub-cases, maybe set up logic in a way that we can easier group things?  This would be a good exercise for you anyway.